linera-core/src/unit_tests/client_tests.rs

```diff
@@ -2250,6 +2252,117 @@ where
     Ok(())
 }
 
+#[test_case(MemoryStorageBuilder::default(); "memory")]
+#[cfg_attr(feature = "storage-service", test_case(ServiceStorageBuilder::new().await; "storage_service"))]
+#[cfg_attr(feature = "rocksdb", test_case(RocksDbStorageBuilder::new().await; "rocks_db"))]
+#[cfg_attr(feature = "dynamodb", test_case(DynamoDbStorageBuilder::default(); "dynamo_db"))]
+#[cfg_attr(feature = "scylladb", test_case(ScyllaDbStorageBuilder::default(); "scylla_db"))]
+#[test_log::test(tokio::test)]
+async fn test_re_propose_fast_block<B>(storage_builder: B) -> anyhow::Result<()>
```

Comment:
To be honest, I don't understand why this test has to be so complicated 🤔 . I thought that we want to test that re-proposing a fast block (originally proposed by superowner) by a regular owner should succeed. It shouldn't require so many steps. Wont' this be enough:?
- we set 3/4 of validators to faulty first
- superowner tries to propose a block in the `Fast` round. It should fail b/c we don't have a quorum of validators
- we set all validators to honest again
- regular owner tries to re-propose it again - it succeeds

Reply:
There is no code path currently where the regular owner would try to finish _another_ owner's proposal unless they also want to make a proposal themselves.

The step where the regular owner first makes an unsuccessful proposal is maybe unnecessary, but the other test (where we re-propose a _regular_ proposal) does it, and I thought it doesn't hurt to make sure in this case, too, that the existence of such a proposal doesn't make the scenario fail.